src/main/java/friday/main/Friday.java

```diff
@@ -0,0 +1,183 @@
+package friday.main;
+
+import java.io.IOException;
+
+import friday.parser.Parser;
+import friday.storage.Storage;
+import friday.task.TaskList;
+import friday.ui.Ui;
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+
+/**
+ * The main class for the Friday task management application.
+ * Friday manages tasks such as Todos, Deadlines, and Events, providing a simple graphical user interface for users.
+ * Users can interact with Friday by entering commands in the text field and receiving responses in the chat area.
+ */
+public class Friday extends Application {
+
+    /** The file path for storing task data. */
+    private static final String DATA_FILE_PATH = "./src/main/java/data/Friday.txt";
+
+    /** The user interface handler for displaying messages and interacting with users. */
+    private final Ui ui;
+
+    /** The storage handler for reading from and writing to the data file. */
+    private final Storage storage;
```

Comment:
Good that your UI and Storage fields are set to final here!

Reply:
thx!